SteamKit2/SteamKit2/Steam/WebAPI/WebAPI.cs

```diff
@@ -222,68 +222,61 @@ public async Task<KeyValue> CallAsync( HttpMethod method, string func, int versi
                     throw new ArgumentNullException( nameof(func) );
                 }
 
-                if ( args == null )
+                if ( args != null && args.TryGetValue( "format", out var format ) )
                 {
-                    args = new Dictionary<string, object>();
+                    if ( !( format is string formatText ) || formatText != "vdf" )
+                    {
+                        throw new ArgumentException( $"{nameof( args )} include unsupported {nameof( format )}: {format}" );
+                    }
                 }
 
-
-                var urlBuilder = new StringBuilder();
                 var paramBuilder = new StringBuilder();
```

Comment:
Is `HttpValueCollection` only in ASP?

Reply:
It seems to be an internal class only in both .NET Framework and Core, and does not appear as an available API anywhere on MS docs site. HttpUtility can parse but not build. FormUrlEncoded content can build but it requires having a full IEnumerable of all the values upfront, and is async.

Reply:
That's lame.